src/content/docs/zh-tw/basics/astro-components.mdx

```diff
@@ -0,0 +1,394 @@
+
+
+## 元件架構
+
+Astro 元件由兩塊組成：**元件腳本**和**元件模板**。兩者各司其職，攜手構築出既容易使用，又能滿足多樣開發需求的框架。
+
+```astro title="src/components/EmptyComponent.astro"
+---
+// 元件腳本（JavaScript）
+---
+<!-- 元件模板（HTML + JS 表達式） -->
+```
+
+### 元件腳本
+
+Astro 使用「程式碼圍欄」（`---`）區別元件腳本。如果你曾寫過 Markdown，你可能知道 *frontmatter*，兩者是相似的概念。Astro 元件腳本的構想就是受此概念啟發。
+
+你可在元件腳本中撰寫渲染模板時需要用到的 JavaScript 程式碼。包含：
+
+- 匯入其他 Astro 元件
+- 匯入其他框架元件，例如 React
+- 匯入資料，例如 JSON 檔
+- 從 API 或資料庫抓取資料
+- 建立變數供模板使用
+
+
+```astro title="src/components/MyComponent.astro"
+---
+import SomeAstroComponent from '../components/SomeAstroComponent.astro';
+import SomeReactComponent from '../components/SomeReactComponent.jsx';
+import someData from '../data/pokemon.json';
+
+// 存取傳入元件的參數，例如 `<X title="Hello, World" />`
+const { title } = Astro.props;
+// 抓取外部資料，從非公開 API 或資料庫都可以
+const data = await fetch('SOME_SECRET_API_URL/users').then(r => r.json());
+---
+<!-- 樣板在這！ -->
+```
+
+程式碼圍欄的設計理念是要「圍住」你寫的 JavaScript，確保它不會跑到前端應用程式、落入使用者手中。在圍欄裡可以安心地寫較耗資源或者敏感的程式碼（例如呼叫你的私有資料庫），不必擔心裡面的程式碼不小心出現在使用者的瀏覽器裡。
+
+:::tip
+元件腳本裡甚至可以寫 TypeScript！
+:::
+
+### 元件模板
+
+元件模板位於程式碼圍欄下方，它決定了最終產出的 HTML。
+
+寫在這裡的 HTML 會被元件渲染，當元件被其他 Astro 頁面匯入使用時亦同。
+
+不只一般的 HTML，[Astro 的元件模板語法](/zh-tw/basics/astro-syntax/) 也支援 **JavaScript 表達式**、Astro [`<style>`](/zh-tw/guides/styling/#styling-in-astro) 和 [`<script>`](/zh-tw/guides/client-side-scripts/#using-script-in-astro) 標籤、**匯入的元件**，以及[**特殊的 Astro 指令**](/zh-tw/reference/directives-reference/)。
+
+```astro title="src/components/MyFavoritePokemon.astro"
+---
+// 這裡是元件腳本！
+import Banner from '../components/Banner.astro';
+import ReactPokemonComponent from '../components/ReactPokemonComponent.jsx';
+const myFavoritePokemon = [/* ... */];
+const { title } = Astro.props;
+---
+<!-- 支援 HTML 註解！ -->
+{/* JS 註解也沒問題！ */}
+
+<Banner />
+<h1>Hello, world!</h1>
+
+<!-- 使用 props 和元件腳本的變數： -->
+<p>{title}</p>
+
+<!-- 透過 `client:` 指令引入其他 UI 框架元件：-->
+<ReactPokemonComponent client:visible />
+
+<!-- 在 HTML 使用 JavaScript 表達式，類似 JSX：-->
+<ul>
+  {myFavoritePokemon.map((data) => <li>{data.name}</li>)}
+</ul>
+
+<!-- 使用模板指令合併 class name，字串甚至是物件都可以！ -->
+<p class:list={["add", "dynamic", {classNames: true}]} />
+```
+
+## 以元件為基礎的設計
+
+元件的設計理念是要能夠**重複使用**和**組合**。你可以在元件裡使用其他元件，創造出更多更複雜的 UI。舉例來說，`Button` 元件可以組合成 `ButtonGroup` 元件：
+
+```astro title="src/components/ButtonGroup.astro"
+---
+import Button from './Button.astro';
+---
+<div>
+  <Button title="Button 1" />
+  <Button title="Button 2" />
+  <Button title="Button 3" />
+</div>
+```
+
+
+## 元件參數
+
+Astro 元件可以自訂和接受參數，供元件模板渲染 HTML 時使用。參數可以在 frontmatter 腳本透過全域的 `Astro.props` 存取。
+
+Here is an example of a component that receives a `greeting` prop and a `name` prop. Notice that the props to be received are destructured from the global `Astro.props` object.
+
+以下範例中，元件接受傳入的 `greeting` 和 `name` 參數。注意參數是從全域的 `Astro.props` 物件解構出來的。
+
+```astro "Astro.props"
+---
+// src/components/GreetingHeadline.astro
+// Usage: <GreetingHeadline greeting="Howdy" name="Partner" />
```

Comment:
It would be interesting to translate the English in the code example into traditional Chinese